### DISCUSSION TITLE
Can we plot discrete pole figures for simulated data?

### DISCUSSION THREAD

**Discussion by yuehengzhang95** [mtex-toolbox/mtex#1338] [Ask Anything] [answered]
I would like to make discrete pole figure plots for the my simulated Laue data. The format the data is [[x1,y1,euler1a,euler1b,euler1c],[x2,y2,euler2a,euler2b,euler2c],[x3,y3,euler3a,euler3b,euler3c],....].x and y are coordinates of the orientations(represented as Euler angles, which is the third, fourth and fifth column - euler1a, euler1b, euler1c etc..). Currently, If I want to plot the pole figures , I have to get the ODF(odf = unimodalODF(ori)) at first, and then we can rotate the data (odf_rotated = rotate(odf, rotation('axis', yvector, 'angle', 90*degree))), finally we can draw the pole figures. 

In our case, I do want to get discrete pole figure plots because there are very characteristic patterns in them that will help us to figure out more information (rotation issues between two reference frames, If we can compare discrete PFs in the same rotational position we should be able to see what’s going on).

I attached my code here for your information, and really appreciate if you can help me!  Many thanks!

%% Specify File Names

fname =  '/laue_results.csv'];

% % create an EBSD variable containing the data
%ebsd = orientation.load(fname,CS,'interface','ang',...'convertEuler2SpatialReferenceFrame','setting 2');

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
cs = crystalSymmetry('432', [4.1 4.1 4.1], 'mineral', 'Gold', 'color', [0.53 0.81 0.98]);
ori = orientation.load(fname,'columnNames',{'x','z','Euler1','Euler2','Euler3'},cs,'convertSpatial2EulerReferenceFrame','setting 2')

odf = unimodalODF(ori)
odf_rotated = rotate(odf, rotation('axis', yvector, 'angle', 90*degree));
odf_rotated_2 = rotate(odf_rotated, rotation('axis', zvector, 'angle', 74*degree));
% odf_rotated_3 = rotate(odf_rotated_2, rotation('axis', zvector, 'angle', 180*degree));
% % 
% % % % % %  
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','intoPlane');
plotPDF(odf_rotated_2,h,'figSize','medium','coordinates','on')
% % % %%
% plotIPDF(odf_rotated_3,[vector3d.X,vector3d.Y,vector3d.Z],'contourf')
% plot(ori)
mtexColorbar

**Comment by kilir** [comment, marked as answer]
Hi,
how about
```
plotPDF(ori,h)
plotIPDF(ori,r)
```
?
Cheers,
Rüdiger

**Reply by yuehengzhang95** [reply]
Hi Rüdiger,

Thanks for your suggestions! I think it works and I have a follow-up questions.
If I would like to apply the rotation to the orientations, what should I do.
I tried the Axis angle parametrisation (specify the rotation axis and the rotation angle) but it does not work. I attached the code and error here. I really appreciate any suggestions!!!


ori_rotated = rotate(ori, rotation('axis', yvector, 'angle', 90*degree));
ori_rotated = rotate(ori_rotated, rotation('axis', zvector, 'angle', 270*degree));

plotPDF(pf_rotated,h)

error:Error using rotate
Too many output arguments.

Error in laue_result_analysis (line 51)
ori_rotated = rotate(ori, rotation('axis', yvector, 'angle', 90*degree));

**Reply by kilir** [reply]
HI,
I think it should be:
```
rot1 = rotation('axis', yvector, 'angle', 90*degree);
rot2 = rotation('axis', zvector, 'angle', 270*degree);

oriR = rot2 * rot1 * ori
```
Cheers,
Rüdiger

**Reply by yuehengzhang95** [reply]
Thanks!

**Reply by yuehengzhang95** [reply]
Hi Rüdiger,

I am curious if we can simulate the 2D orientation map with inverse pole figure color for my simulated dataset? The purpose is for me to compare the EBSD map with my simulated data.
The format the simulated data is [[x1,y1,euler1a,euler1b,euler1c],[x2,y2,euler2a,euler2b,euler2c],[x3,y3,euler3a,euler3b,euler3c],....].x and y are coordinates of the orientations(represented as Euler angles, which is the third, fourth and fifth column - euler1a, euler1b, euler1c etc..).

PS. if we want to plot the map based on the sample x direction(010) instead of Z direction(001), what else do I need to do. Many thanks!

Yueheng
